weekly-scan/template.yaml

```diff
@@ -0,0 +1,145 @@
+                    def image_in_registry = false
+                    def success = false
+                    try{
+                        stage('Check if image exists in registry'){
+                            sh "curl ${image_tags} | grep ${DESIRED_TAG}"
+                            image_in_registry = true
+                        }
+                        stage('Pull the docker image'){
+                            sh "docker pull ${image_name_with_registry}"
+                        }
+                        stage('Scan the image') {
+                            parallel (
+                                "RPM updates": {
+                                  sh (returnStdout: true, script: "docker run --rm --volumes-from `docker ps -qf name=${daemonset_name}` --entrypoint /bin/python ${image_name} /opt/scanning/yumupdates.py > yum-check-update 2>&1")
+                                  sh "cat yum-check-update"
+                                },
+                                "Verify RPMs": {
+                                  sh (returnStdout: true, script: "docker run --rm --volumes-from `docker ps -qf name=${daemonset_name}` --entrypoint /bin/python ${image_name} /opt/scanning/rpmverify.py > rpm-verify 2>&1")
+                                  sh "cat rpm-verify"
+                                },
+                                "Miscellaneous updates": {
+                                  sh (returnStdout: true, script: "docker run --rm --volumes-from `docker ps -qf name=${daemonset_name}` --entrypoint /bin/python ${image_name} /opt/scanning/misc_package_updates.py all > misc-updates 2>&1")
+                                  sh "cat misc-updates"
+                                },
+                                "Container capabilities": {
+                                  def run_label = sh (script: "docker inspect ${image_name} --format '{{ index .Config.Labels \"RUN\" }}'")
+                                  sh (returnStdout: true, script: "docker run --rm --volumes-from `docker ps -qf name=${daemonset_name}` --entrypoint /bin/python ${image_name} /opt/scanning/container-capabilities.py ' ${run_label} ' > capabilities 2>&1")
+                                  sh "cat capabilities"
+                                }
+                            )
+                        }
+                        stage('Remove the docker image'){
+                            sh "docker rmi ${image_name_with_registry}"
+                            success = true
+                        }
+                    }
+                    finally {
+                        if (image_in_registry==true) {
+                            if (success==true) {
+                                sh "echo -e 'Weekly scan for ${image_name_with_registry} was successful' | mail -r ${FROM_ADDRESS} -S smtp=${SMTP_SERVER} -s 'SUCCESS: Weekly scan for ${image_name_with_registry} is complete' ${NOTIFY_EMAIL} "
```

Comment:
(commenting after next comment)

In fact, let's remove the registry name altogether from the subject. The `registry.centos.org` will be redundant and makes subject line long.

Reply:
Sorted